@shlevy 

```
src/libstore/build.cc:913:17: error: constructor for 'nix::DerivationGoal' must explicitly initialize the member 'autoDelSandbox' which does not have a default constructor
DerivationGoal::DerivationGoal(const Path & drvPath, const BasicDerivation & drv,
                ^
src/libstore/build.cc:785:16: note: member is declared here
    AutoDelete autoDelSandbox;
               ^
src/libutil/util.hh:196:7: note: 'nix::AutoDelete' declared here
class AutoDelete
      ^
```

Unfamiliar with C++, not sure what to do here.

--- a/src/libstore/build.cc
+++ b/src/libstore/build.cc
@@ -2534,7 +2534,7 @@ void DerivationGoal::runChild()
 
             Path sandboxFile = drvPath + ".sb";
             if (pathExists(sandboxFile)) deletePath(sandboxFile);
-            autoDelSandbox = AutoDelete(sandboxFile);
+            autoDelSandbox.reset(sandboxFile, false);
 
             writeFile(sandboxFile, sandboxProfile);
 

--- a/src/libutil/util.cc
+++ b/src/libutil/util.cc
@@ -629,7 +629,7 @@ void AutoDelete::cancel()
 }
 
 void AutoDelete::reset(const Path & p, bool recursive = true) {
-    this-> p = p;
+    path = p;
     this->recursive = recursive;
     del = true;
 }

--- a/src/libutil/util.cc
+++ b/src/libutil/util.cc
@@ -599,6 +599,8 @@ string drainFD(int fd)
 //////////////////////////////////////////////////////////////////////
 
 
+AutoDelete::AutoDelete() : del{false} {}
+
 AutoDelete::AutoDelete(const string & p, bool recursive) : path(p)
 {
     del = true;
@@ -624,6 +626,12 @@ AutoDelete::~AutoDelete()
 void AutoDelete::cancel()
 {
     del = false;
+}
+
+void AutoDelete::reset(const Path & p, bool recursive = true) {
+    this-> p = p;
+    this->recursive = recursive;
+    del = true;
 }
 
 

--- a/src/libutil/util.cc
+++ b/src/libutil/util.cc
@@ -628,7 +628,7 @@ void AutoDelete::cancel()
     del = false;
 }
 
-void AutoDelete::reset(const Path & p, bool recursive = true) {
+void AutoDelete::reset(const Path & p, bool recursive) {
     path = p;
     this->recursive = recursive;
     del = true;

--- a/src/libutil/util.hh
+++ b/src/libutil/util.hh
@@ -199,9 +199,11 @@ class AutoDelete
     bool del;
     bool recursive;
 public:
+    AutoDelete();
     AutoDelete(const Path & p, bool recursive = true);
     ~AutoDelete();
     void cancel();
+    void reset(const Path & p, bool recursive = true);
     operator Path() const { return path; }
 };
 

